src/openforms/formio/migration_converters.py

```diff
@@ -0,0 +1,51 @@
+"""
+Expose a centralized registry of migration converters.
+
+This registry is used by the data migrations *and* form import. It guarantees that
+component definitions are rewritten to be compatible with the current code.
+"""
+from typing import Protocol
+
+from .typing import Component
+
+
+class Converter(Protocol):
+    def __call__(self, component: Component) -> bool:  # pragma: no cover
```

Comment:
Why not
```suggestion
    def __call__(self, component: Component) -> Component:  # pragma: no cover
```
That will give it nice properties like composability.

And in place mutation closes the door to parallelization.

Reply:
there is a has changed/has not changed flag required for the data migration so it doesn't perform an update on `FormDefinition` records that weren't actually updated.

> And in place mutation closes the door to parallelization.

does it really? in a data migration you can still parallellize the queryset itself and run the converter in parallel for every form definition in the queryset, at import time you can still do something similar for every FD in the import data.

If  you have a 1000 components in a single form definition and that would *require* parallel conversion, then I think you're building shitty user hostile forms that are way too long too.

Reply:
> And in place mutation closes the door to parallelization.

I don't know when the GILectomy will land, but I heard they're aiming for the year of the Linux desktop.